### PR TITLE
fix(server): Exclude archived assets from search-explore #4041

### LINKED ISSUE #4041: [BUG] Archived photos still showing in Explore/Places
### The bug

When I archive a photo, it gets hidden from my Photo Timeline like i expect, but when I go to the Explore tab, they are still showing up under Places.  This seems to me to be a bug

### The OS that Immich Server is running on

Ubuntu 22.04.3 LTS (Docker)

### Version of Immich Server

1.77.0

### Version of Immich Mobile App

1.77.0 build.100

### Platform with the issue

- [ ] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: [ "start.sh", "immich" ]
    environment:
      DB_HOSTNAME: postgresql-immich
      DB_USERNAME: pguser-immich
      DB_PASSWORD: XXXXXXXXXXXXXXXXXXXXX
      DB_DATABASE_NAME: pgdb-immich
      DB_PORT: 5432
      UPLOAD_LOCATION: /usr/src/app/upload
      IMMICH_WEB_URL: http://immich-web:3000
      IMMICH_SERVER_URL: http://immich-server:3001
      IMMICH_MACHINE_LEARNING_URL: http://immich-machine-learning:3003
      REDIS_HOSTNAME: redis
      REDIS_PASSWORD: 'XXXXXXXXXXXXXXXXXXXXXXX'
      TYPESENSE_API_KEY: XXXXXXXXXXXXXXXXXXXXXXXXX
      TYPESENSE_HOST: typesense
    ports:
      - 3001:3001
    volumes:
      - ${DOCKERCONFDIR}/immich:/data
      - ${DOCKERSTORAGEDIR}:/storage
      - /media/USB2/photos:/usr/src/app/upload
      - /media/USB2/imported_photos:/media/USB2/imported_photos:ro
    depends_on:
      - redis
      - postgresql-immich
      - typesense
    restart: always
```

### Your .env content

```Shell
environment:
      DB_HOSTNAME: postgresql-immich
      DB_USERNAME: pguser-immich
      DB_PASSWORD: XXXXXXXXXXXXXXXXXXXXX
      DB_DATABASE_NAME: pgdb-immich
      DB_PORT: 5432
      UPLOAD_LOCATION: /usr/src/app/upload
      IMMICH_WEB_URL: http://immich-web:3000
      IMMICH_SERVER_URL: http://immich-server:3001
      IMMICH_MACHINE_LEARNING_URL: http://immich-machine-learning:3003
      REDIS_HOSTNAME: redis
      REDIS_PASSWORD: 'XXXXXXXXXXXXXXXXXXXXXXX'
      TYPESENSE_API_KEY: XXXXXXXXXXXXXXXXXXXXXXXXX
      TYPESENSE_HOST: typesense
```

### Reproduction steps

```bash
1. Take a new picture with my phone
2. Upload picture to immich
3. Locate picture in Timeline
4. select picture
5. choose Archive (photo disappears from timeline)
6. Navigate to Explore
7. Under places are, find the "location" that you took the picture
8. verify that the archived photo shows in this location (I would think it shouldn't)
...
```

### Additional information

Unless I am thinking this through incorrectly, I assume that are archived photo will not show up anywhere besides the Archive album

### PR DESCRIPTION
**Pull request by sellnat77**
## Description
  - Excludes assets with the `.archived` from the /search/explore route 

Fixes #4041

## How Has This Been Tested?

- [x] Added unit test to verify archived assets arent returned in the explore page
- [x] Uploaded image with a place -> went to explore -> verify image shows up in explore -> archive image -> go to explore -> verify image no longer shows up in places

## Screenshots (if appropriate):
Pre archive
Screenshot from 2023-09-17 10-37-11
Post archive
Screenshot from 2023-09-17 10-38-10

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if applicable

### CODE REVIEW

**Review by jrasm91 — COMMENTED** [review]
I think this returns one asset for each example. If the "image" asset is archived, this would hide the whole category, even if there are other assets tagged "image" that are not archived.

We probably want to update the typesense repository implementation for this.

**Comment by sellnat77** [comment]
> I think this returns one asset for each example. If the "image" asset is archived, this would hide the whole category, even if there are other assets tagged "image" that are not archived.
> 
> We probably want to update the typesense repository implementation for this.

Would that be a change deeper at this level?
https://github.com/immich-app/immich/blob/main/server/src/domain/search/search.service.ts#L152
or...im not sure where to find 'typesense repository'? like more of a DB/migration type of change?

**Comment by jrasm91** [comment]
Probably add it here:

https://github.com/immich-app/immich/blob/main/server/src/infra/repositories/typesense.repository.ts#L153

File: server/src/infra/typesense-schemas/asset.schema.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { CollectionCreateSchema } from 'typesense/lib/Typesense/Collections';
 
-export const assetSchemaVersion = 9;
+export const assetSchemaVersion = 10;
```
Comment (sellnat77) [inline server/src/infra/typesense-schemas/asset.schema.ts:3]: Making an educated guess that i need to bump this version number. Ive never worked with typesense before so please someone correct me if this is wrong!

Comment (jrasm91) [reply server/src/infra/typesense-schemas/asset.schema.ts:3]: Yup, this looks good. I'll try to test it later.

**Review by jrasm91 — APPROVED** [review]

**Pull request merged**